alchemy/src/neon/endpoint.ts

```diff
@@ -0,0 +1,689 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { handleApiError } from "./api-error.ts";
+import { createNeonApi, type NeonApiOptions, type NeonApi } from "./api.ts";
+
+/**
+ * Properties for creating or updating a Neon endpoint
+ */
+export interface NeonEndpointProps extends NeonApiOptions {
+  /**
+   * The ID of the project containing the branch
+   */
+  project_id: string;
+
+  /**
+   * The ID of the branch where the endpoint will be created
+   */
+  branch_id: string;
+
+  /**
+   * Type of endpoint (read-write or read-only)
+   */
+  type: "read_write" | "read_only";
+
+  /**
+   * Compute provisioner for the endpoint
+   */
+  compute_provisioner?: "k8s-pod" | "k8s-neonvm";
+
+  /**
+   * PostgreSQL settings for the endpoint
+   */
+  settings?: {
+    pg_settings?: Record<string, string>;
+  };
+
+  /**
+   * Whether connection pooling is enabled
+   * @default false
+   */
+  pooler_enabled?: boolean;
+
+  /**
+   * Connection pooling mode
+   * @default "transaction"
+   */
+  pooler_mode?: "session" | "transaction";
+
+  /**
+   * Whether the endpoint is disabled
+   * @default false
+   */
+  disabled?: boolean;
+
+  /**
+   * Whether passwordless access is enabled
+   * @default false
+   */
+  passwordless_access?: boolean;
+
+  /**
+   * Suspend timeout in seconds
+   */
+  suspend_timeout_seconds?: number;
+
+  /**
+   * Provisioner for the endpoint
+   */
+  provisioner?: "k8s-pod" | "k8s-neonvm";
+
+  /**
+   * Region ID for the endpoint
+   */
+  region_id?: string;
+
+  /**
+   * Whether to adopt an existing endpoint if it already exists
+   * @default false
+   */
+  adopt?: boolean;
+}
+
+/**
+ * Response structure for endpoint API operations
+ */
+export interface NeonEndpointType {
```

Comment:
This is an internal type that is not needed outside of this file. Inline if possible. Move to bottom where the utility api functions are otherwise.